Readme.md

```diff
@@ -68,6 +68,18 @@ composer test-all      # psalm, compiler & core tests after each other
 > composer test-core
 ```
 
+### Architecture Decision Record (ADR) 
+
+These helps us to bring more structure to our code. However, the most important thing with ADRs or RFCs is the process
+on how they are created, discussed, processed and so forth. Just adding ADR file will not make a difference. 
+Additionally, we need a way to track if all ADRs are full filled when new code is added or changed.
+
+#### Adding an ADR Process
+
```

Comment:
Add an introduction: "In order to make a substantial change it is a good practice to discuss the idea before implementing it. ..." (or similar)